minio/credentials/providers.py

```diff
@@ -98,11 +98,11 @@ def __init__(
         query_params = {
             "Action": "AssumeRole",
             "Version": "2011-06-15",
-            "DurationSeconds": str(
+            "DurationSeconds": str(int(
```

Comment:
The actual fix is
```py
diff --git a/minio/credentials/providers.py b/minio/credentials/providers.py
index 00d73b8..1447a59 100644
--- a/minio/credentials/providers.py
+++ b/minio/credentials/providers.py
@@ -37,9 +37,9 @@ from minio.xml import find, findtext
 
 from .credentials import Credentials
 
-_MIN_DURATION_SECONDS = timedelta(minutes=15).total_seconds()
-_MAX_DURATION_SECONDS = timedelta(days=7).total_seconds()
-_DEFAULT_DURATION_SECONDS = timedelta(hours=1).total_seconds()
+_MIN_DURATION_SECONDS = int(timedelta(minutes=15).total_seconds())
+_MAX_DURATION_SECONDS = int(timedelta(days=7).total_seconds())
+_DEFAULT_DURATION_SECONDS = int(timedelta(hours=1).total_seconds())
 
 
 def _parse_credentials(data, name):
```

Reply:
Fine by me, of course that assumes the user passes an int. If they pass a string the comparison won't work, if they pass a float the same issue comes up. 

Personally I'd be in favor of actually using what ever the user provides - even if it is less than the default, only falling back to the default if a value hasn't been provided.

```python
duration_seconds
if duration_seconds > _DEFAULT_DURATION_SECONDS
else _DEFAULT_DURATION_SECONDS
```

Reply:
If you worry about `duration_seconds` argument type, you can add a check to raise `ValueError` if it is not `int`. That is the right way to fix.